cypress/scripts/anvil.ts

```diff
@@ -1,21 +1,64 @@
-/* eslint-disable sonarjs/no-duplicate-string */
-import { spawn } from "child_process";
-import { getFastestRpcUrl } from "../../shared/helpers";
+import { spawnSync } from "child_process";
+import { useHandler } from "../../static/scripts/rewards/web3/use-rpc-handler";
+import { RPCHandler } from "@ubiquity-dao/rpc-handler";
 
-async function forkNetwork() {
-  const fastestRpcUrl = await getFastestRpcUrl(100);
+class Anvil {
+  rpcs: string[] = [];
+  rpcHandler: RPCHandler | null = null;
+  networkIdToFork = 100;
+  networkIdForAnvilToForkAs = 31337;
 
-  const anvil = spawn("anvil", ["--chain-id", "31337", "--fork-url", fastestRpcUrl, "--host", "127.0.0.1", "--port", "8545"], {
-    stdio: "inherit",
-  });
+  async init() {
+    this.rpcHandler = useHandler(this.networkIdToFork);
+    console.log(`[RPCHandler] Fetching RPCs...`);
+    await this.rpcHandler.testRpcPerformance();
```

Comment:
Were we not supposed to use `getFirstAvailableRpcProvider` as per the specifications of the issue.

Reply:
1. It's still possible for anvil to panic and crash unrelated to the RPC so we'd need some retry logic anyway, this also happened to me during testing.
2. The function returned a wss provider in error during my testing so the fn needed fixed